apps/studio/src/features/editing-experience/hooks/useTextEditor.ts

```diff
@@ -34,7 +35,33 @@ export interface BaseEditorProps {
 }
 
 const BASE_EXTENSIONS: Extensions = [
-  Link.configure({
+  Link.extend({
+    addProseMirrorPlugins() {
+      return [
+        // NOTE: This plugin is used to transform links inside the HTML content
+        // copied by the user from the preview, as browsers will automatically
+        // transform relative links (in the form of [resource:siteId:resourceId])
+        // into absolute links:
+        // https://studio.isomer.gov.sg/sites/1/pages/[resource:siteId:resourceId]
+        // This plugin will transform the absolute links back into the relative
+        // links, so that the original link is preserved in the editor.
+        new Plugin({
+          key: new PluginKey("transformReferenceLinks"),
+          props: {
+            transformPastedHTML(html, _) {
+              return html.replaceAll(
+                /href="(?:http|https):\/\/[^[]*\/\[resource:(\d+):(\d+)\]"/g,
+                (_, siteId, resourceId) => {
+                  return `href="[resource:${siteId}:${resourceId}]"`
+                },
```

Comment:
should we 
1. put this into a util so we can write basic unit test for it?
2. extract `[resource:(\d+):(\d+)\]` into a const and import it instead? seems to be used in a number of places

Reply:
Updated in ced21bd362465da4508444f738b092a364f9d766. Didn't create the tests yet as I think we haven't decided how we want to do testing for the editing experience feature.